File: web/src/types/system.ts
export interface HealthResponse {
  status: string
  time_ms: number
}

export interface SystemConfigResponse {
  registration_enabled: boolean
  btc_eth_leverage: number
  altcoin_leverage: number
  runtime_alert_webhook_enabled: boolean
  runtime_alert_webhook_auth_header_set: boolean
  runtime_alert_webhook_timeout_secs: number
  runtime_alert_webhook_max_retries: number
  runtime_alert_webhook_retry_backoff_ms: number
  runtime_alert_webhook_signing_enabled: boolean
  runtime_alert_webhook_signing_header_set: boolean
  runtime_alert_webhook_signing_timestamp_header_set: boolean
  runtime_alert_webhook_signing_max_age_secs: number
}

File: core/src/contracts/system.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
pub struct HealthResponse {
    pub status: &'static str,
    pub time_ms: u128,
}

#[derive(Debug, Serialize)]
pub struct SystemConfigResponse {
    pub registration_enabled: bool,
    pub btc_eth_leverage: u32,
    pub altcoin_leverage: u32,
    pub runtime_alert_webhook_enabled: bool,
    pub runtime_alert_webhook_auth_header_set: bool,
    pub runtime_alert_webhook_timeout_secs: u64,
    pub runtime_alert_webhook_max_retries: u64,
    pub runtime_alert_webhook_retry_backoff_ms: u64,
    pub runtime_alert_webhook_signing_enabled: bool,
    pub runtime_alert_webhook_signing_header_set: bool,
    pub runtime_alert_webhook_signing_timestamp_header_set: bool,
    pub runtime_alert_webhook_signing_max_age_secs: u64,
}
